serveradmin/backup.py

```
@@ -0,0 +1,96 @@
+import os
+import shutil
+import zipfile
+
+from datetime import datetime
+
+from django.conf import settings
+
+
+from ideasbox import __version__
+
+
+def make_name():
+    """Return backup formatted file name."""
+    return '{}.zip'.format('_'.join([
+        settings.IDEASBOX_ID,
+        __version__,
+        datetime.now().strftime(Backup.DATE_FORMAT)
+    ]))
+
+
+class Backup(object):
+
+    FORMAT = 'zip'
+    DATE_FORMAT = "%Y%m%d%H%M"
+    ROOT = os.path.join(settings.STORAGE_ROOT, 'backups')
+
+    def __init__(self, name):
+        if not name.endswith('.zip'):
+            raise ValueError
```

You should raise a specific error here.